src/ddmd/expressionsem.d

```diff
@@ -3267,6 +3267,9 @@ private extern (C++) final class ExpressionSemanticVisitor : Visitor
         }
         if (exp.e1.op == TOKslice || exp.e1.type.ty == Tarray || exp.e1.type.ty == Tsarray)
         {
+            if (exp.e1.op == TOKslice)
+                (cast(SliceExp)exp.e1).arrayop = true;
```

Comment:
I hope we can remove that special case once the library arrayOps story is merged. Still need to make it scope safe though.